### PR TITLE
Enable capturing triage dump on test crash

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alan.jowett@microsoft.com>

## Description

Use Windows Error Reporting to capture triage dump when a test crashes.

## Testing

Manually verified.

## Documentation

Updated docs for setting up the self-host runner.

Resolves: #991

### CODE REVIEW

File: docs/SelfHostedRunnerSetup.md [hunk 1]
```diff
@@ -38,4 +38,5 @@ This document discusses the steps to set up such a selfhosted actions-runner tha
     3) $env:Path += ";C:\Program Files\ProcDump"
     4) Create two directories c:\dumps\x64\Release and c:\dumps\x64\Debug
     5) Run: `procdump64.exe -accepteula -i -r -ma <path>` once for each of the two directories above, substituting `path` for the directory path.
+11) Setup Windows Error Reporting [Local Dump Collection](https://docs.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps).
```
Comment (dthaler) [inline docs/SelfHostedRunnerSetup.md:41]:
```suggestion
11) Set up Windows Error Reporting [Local Dump Collection](https://docs.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps).
```

File: tests/libs/util/wer_report.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        vectored_exception_handler_handle = AddVectoredExceptionHandler(TRUE, _wer_report::vectored_exception_handler);
+    }
+    ~_wer_report()
+    {
+        if (vectored_exception_handler_handle)
+            RemoveVectoredExceptionHandler(vectored_exception_handler_handle);
+    }
+
+  private:
+    static constexpr const char environment_variable_name[] = "EBPF_ENABLE_WER_REPORT";
```
Comment (dthaler) [inline tests/libs/util/wer_report.cpp:44]: https://github.com/microsoft/ebpf-for-windows/blob/main/docs/DevelopmentGuide.md#naming-conventions says:
> Prefix names with `_` to indicate internal and private fields or methods (e.g., `_internal_field`, `_internal_method()`).

File: .github/workflows/reusable-test.yml [hunk 3]
```diff
@@ -96,6 +96,9 @@ jobs:
         cd c:/dumps/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}
         echo "test" > c:/dumps/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}/test.txt
         procdump64.exe -accepteula -i -r -ma
+        New-Item -Path "HKLM:\SOFTWARE\Microsoft\Windows\Windows Error Reporting\LocalDumps" -ErrorAction SilentlyContinue
```
Comment (shankarseal) [inline .github/workflows/reusable-test.yml:99]: is the procdump changes needed any longer if we are setting the below regkeys?

File: tests/libs/util/wer_report.h [hunk 4]
```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation
```
Comment (shankarseal) [inline tests/libs/util/wer_report.h:1]: Does every test app link to this library?

Comment (Alan-Jowett) [reply tests/libs/util/wer_report.h:1]: This is included in catch_wrapper.hpp, which is used by every test.

File: tests/libs/util/catch_wrapper.hpp [hunk 5]
```diff
@@ -18,3 +18,5 @@
 #include "catch2/catch_all.hpp"
 #endif
 #pragma warning(pop)
+
+#include "wer_report.h"
```
Comment (shankarseal) [inline tests/libs/util/catch_wrapper.hpp:22]: nit: consider rename wer_report.hpp

**Review by shankarseal — APPROVED** [review]

**Comment by Alan-Jowett** [comment]
Will attempt to improve codecoverage on the wer_report code in a future PR.

**Review by shankarseal — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**